src/SmartTransactionsController.ts

```diff
@@ -590,9 +606,12 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
         nonce: incrementNonceInHex(unsignedApprovalTransactionWithNonce.nonce),
       };
     } else {
-      unsignedTradeTransactionWithNonce = await this.addNonceToTransaction(
-        tradeTx,
-      );
+      unsignedTradeTransactionWithNonce = tradeTx;
+      if (!unsignedTradeTransactionWithNonce.nonce) {
+        unsignedTradeTransactionWithNonce = await this.addNonceToTransaction(
+          unsignedTradeTransactionWithNonce,
+        );
+      }
```

Comment:
Not a huge deal, but this reads a little odd to me. Var name is `withNonce` but it might not exist on the object. Is it possible to do the following: 

```suggestion
      if (!unsignedTradeTransactionWithNonce.nonce) {
        unsignedTradeTransactionWithNonce = await this.addNonceToTransaction(
          tradeTx,
        );
      }
```

Reply:
Agreed, but we still need to set the `tradeTx` into `unsignedTradeTransactionWithNonce` if it has a nonce.

I've done this:
```
else if (tradeTx.nonce) {
      unsignedTradeTransactionWithNonce = tradeTx;
    } else {
      unsignedTradeTransactionWithNonce = await this.addNonceToTransaction(
        tradeTx,
      );
    }
```